File: 99-bottles/99-bottles.go
package main

import "fmt"

func singTheSong(b int) {
	if b > 1 {
		fmt.Printf("%v bottles of beer on the wall, %v bottles of beer.\n",
			b, b)
		fmt.Printf("Take one down and pass it around, %v bottles of beer on the wall.\n", b-1)
	} else if b == 1 {
		fmt.Printf("1 bottle of beer on the wall, 1 bottle of beer.\n")
		fmt.Printf("Take one down and pass it around, no more bottles of beer on the wall.\n")
	} else {
		fmt.Printf("No more bottles of beer on the wall, no more bottles of beer.\n")
		fmt.Printf("Go to the store and buy some more, 99 bottles of beer on the wall.\n")
	}
}

func main() {
	for b := 99; b >= 0; b-- {
		singTheSong(b)
	}
}

File: 99-bottles/99-bottles.c
#include <stdio.h>

void singTheSong(int b){
	if (b > 1) {
		printf("%d bottles of beer on the wall, %d bottles of beer.\n",
			b, b);
		printf("Take one down and pass it around, %d bottles of beer on the wall.\n", b-1);
	} else if (b == 1) {
		printf("1 bottle of beer on the wall, 1 bottle of beer.\n");
		printf("Take one down and pass it around, no more bottles of beer on the wall.\n");
	} else {
		printf("No more bottles of beer on the wall, no more bottles of beer.\n");
		printf("Go to the store and buy some more, 99 bottles of beer on the wall.\n");
	}
}

int main(int argc, char** argv){
    for (int b = 99; b >= 0; --b){
        singTheSong(b);
		printf("\n");
    }
    return 0;
}

